### ISSUE TITLE
about gradient mismatch hypothesis

### ISSUE THREAD

**Issue by payphone131** [git-disl/Virus#2]
Hi, I found your paper really interesting, but there is still something that I do not quite understand. Does the "gradient mismatch" refer to the phenomenon that the gradients of guardrail attack are different from the gradients of mixing attack while fintuning? If so, how to compute the gradients of mixing attack for the part of training data that fail to pass the guardrail? To my understanding, training data that fail to pass the guardrail cannot be used during finetuning, which means they don't have gradients.

**Comment by huangtiansheng** [comment]
Thanks for reaching out. The Virus method is applied before the real fine-tuning attack happens. We assume the attacker has whitebox access to the guardrail model and the victim model. 

In this case, the attacker can access the gradient of the mixing data and the gradient of the opmizable data over the **victim model**, and introduce the gradient similarity loss term. The guardrail model is not involved in the calculation of gradient similarity loss. 

Let me know if it does not address your question.

**Comment by payphone131** [comment]
So to my understanding, what you said means that when calculating the gradient similarity loss, the guardrail model is not in use, so that no data are filtered out and every piece of data can be used to calculate the gradient similarity loss. The Virus method will optimize the input data, making them able to pass the guardrail and maintain harmfulness simultaneously.

**Comment by huangtiansheng** [comment]
Yes, your understanding is perfectly correct. The guardrail model is only used in the jailbreak loss to enable the data to pass the guardrail. In this loss, we need the weight of a guardrail model, but not the victim LLM.   

For the gradient similarity term, we use the full data (without filtering) to enable the data to achieve harmfulness. To achieve this loss, we need the weight of victim LLM, but not needing the guardrail model.

**Comment by payphone131** [comment]
Thanks, your reply really solved my question.
